core/src/main/scala/pureconfig/DerivedReaders.scala

```diff
@@ -40,17 +40,7 @@ trait DerivedReaders extends DerivedReaders1 {
 /**
  * Trait containing `ConfigReader` instances for collection, product and coproduct types.
  */
-trait DerivedReaders1 {
-
-  private[pureconfig] trait WrappedConfigReader[Wrapped, SubRepr] extends ConfigReader[SubRepr]
-
-  private[pureconfig] trait WrappedDefaultValue[Wrapped, SubRepr <: HList, DefaultRepr <: HList] {
-    def fromWithDefault(config: ConfigValue, default: DefaultRepr): Either[ConfigReaderFailures, SubRepr] = config match {
-      case co: ConfigObject => fromConfigObject(co, default)
-      case other => fail(WrongType(other.valueType, Set(ConfigValueType.OBJECT), ConfigValueLocation(other), ""))
-    }
-    def fromConfigObject(co: ConfigObject, default: DefaultRepr): Either[ConfigReaderFailures, SubRepr]
-  }
+trait DerivedReaders1 extends DerivedReaders2 {
```

Comment:
@leifwickland After the latest algorithm revamp, I think this separation isn't needed anymore (the macro seeing them as equal was a smell a of a bigger problem which I had to fix). I think I'm going to revert this in order to reduce complexity here.